tests/foreman/api/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,288 @@
+        """
+
+        assert module_azurerm_finishimg.architecture.id == module_architecture.id
+        assert module_azurerm_finishimg.compute_resource == module_azurerm_cr
+        assert module_azurerm_finishimg.username == settings.azurerm.username
+        assert module_azurerm_finishimg.uuid == AZURERM_RHEL7_FT_IMG_URN
+
+    @upgrade
+    @tier2
+    def test_positive_create_cloud_init_image(self,
+                                              module_azurerm_cloudimg,
+                                              module_azurerm_cr,
+                                              module_architecture):
+        """Cloud Init template image along with username is being added in AzureRM CR
+
+        :id: 05ea1b20-0dfe-4af3-b1b7-a82604aa1e79
+
+        :steps:
+            1. Create a AzureRM Compute Resource.
+            2. Add a Cloud init supported image in it.
+
+        :expectedresults: Cloud init image should be added in AzureRM CR along with username
+
+        :CaseLevel: Integration
+        """
+
+        assert module_azurerm_cloudimg.architecture.id == module_architecture.id
+        assert module_azurerm_cloudimg.compute_resource.id == module_azurerm_cr.id
+        assert module_azurerm_cloudimg.username == settings.azurerm.username
+        assert module_azurerm_cloudimg.uuid == AZURERM_RHEL7_UD_IMG_URN
+
+    @upgrade
+    @tier2
+    def test_positive_check_available_networks(self, azurermclient, module_azurerm_cr):
+        """Check networks from AzureRM CR are available to select during host provision.
+
+        :id: aee5d077-668e-4f4d-adee-6472f0e4ecaa
+
+        :expectedresults: All the networks from AzureRM CR should be available.
+
+        :CaseLevel: Integration
+        """
+
+        cr_nws = module_azurerm_cr.available_networks()
+        portal_nws = azurermclient.list_network()
+        assert len(portal_nws) == len(cr_nws['results'])
+
+
+class TestAzureRMHostProvisioningTestCase:
+    """ AzureRM Host Provisioning Tests
+
+    """
+    @pytest.fixture(scope='class', autouse=True)
+    def class_setup(self, request, module_domain, module_azurerm_cr):
+        """
+        Sets Constants for all the Tests, fixtures which will be later used for assertions
+        """
+        request.cls.region = settings.azurerm.azure_region
+        request.cls.rhel7_ft_img = AZURERM_RHEL7_FT_IMG_URN
+        request.cls.rg_default = AZURERM_RG_DEFAULT
+        request.cls.premium_os_disk = AZURERM_PREMIUM_OS_Disk
+        request.cls.platform = AZURERM_PLATFORM_DEFAULT
+        request.cls.vm_size = AZURERM_VM_SIZE_DEFAULT
+        request.cls.hostname = gen_string('alpha')
+        request.cls.fullhostname = '{}.{}'.format(self.hostname, module_domain.name).lower()
+
+        request.cls.compute_attrs = {
+            "resource_group": self.rg_default,
+            "vm_size": self.vm_size,
+            "username": settings.azurerm.username,
+            "password": settings.azurerm.password,
```

Comment:
I think this could be randomly generated, need not to pass ot from properties.

Reply:
done.